main/internal/implementations/terraform_value_objects/terraformer_value_objects.go

```diff
@@ -0,0 +1,39 @@
+package terraformValueObjects
+
+import (
+	"fmt"
+	"strings"
+)
+
+type CloudRegion string
+
+type CloudRegionsDecoder []CloudRegion
+
+func (d *CloudRegionsDecoder) Decode(value string) error {
+	*d = []CloudRegion{}
+	if strings.Trim(value, " ") == "" || strings.Trim(value, " ") == "[]" {
+		return nil
+	}
+
+	if value[0] != '[' || value[len(value)-1] != ']' {
+		return fmt.Errorf("the value %v is not a list", value)
+	}
+
+	value = value[1 : len(value)-1]
+	regionsList := strings.Split(value, ",")
+
+	for _, region := range regionsList {
```

Comment:
Let's also validate that each region is a valid region from the known lists of one of the big three cloud providers. That way we can fail early in the execution and give a helpful error message